### ISSUE TITLE
Possible bug in Flying Saucers 'Graphics2DRenderer'

### ISSUE THREAD

**Issue by skjardenCode** [danfickle/openhtmltopdf#63]
Hi Daniel,

thanks a lot for your ongoing efforts on this helpful project!

We recently came up with the use case to render a certain small HTML to an image before we render it to PDF. As OpenHtmlToPdf is based on Flying Saucer, we tried to use the ```com.openhtmltopdf.simple.Graphics2DRenderer``` class to do so, which should be an easy task.

Unfortunately (using Java 6 and Windows 10 with the latest OpenHtmlToPdf version available in Maven Central), the image output of ```Graphics2DRenderer``` yield wrong results, mainly what seems like an encoding problem.

Please, see the following simple example code:

```java
String xhtml = 
        "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Strict//EN\" "
                + "\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\">"
        + "<html xmlns=\"http://www.w3.org/1999/xhtml\" xml:lang=\"de\" lang=\"de\">"
            + "<body>"
                + "Moonbase"
            + "</body>"
        + "</html>";

InputSource inputSource = new InputSource( new BufferedReader( new StringReader( xhtml ) ) );
Document document = XMLResource.load( inputSource ).getDocument();

Graphics2DRenderer graphics2dRenderer = new Graphics2DRenderer();
graphics2dRenderer.setDocument( document, null );

BufferedImage bufferedImage = new BufferedImage( 500, 500, BufferedImage.TYPE_INT_RGB );
Graphics2D g2d = bufferedImage.createGraphics();

graphics2dRenderer.layout( g2d, new Dimension( 500, 500 ) );
graphics2dRenderer.render( g2d );

g2d.dispose();

File outputfile = File.createTempFile( Graphics2DRendererTest.class.getSimpleName(), ".png" );
ImageIO.write( bufferedImage, "png", outputfile );
Desktop.getDesktop().open( outputfile );
```

The image you get (trying to render the word "Moonbase") is the following:

![image](https://cloud.githubusercontent.com/assets/23582335/21961844/2294ffc0-db15-11e6-8f2e-df7f942aafad.png)

We than integrated the latest Flying Saucer version into our project, next to OpenHtmlToPdf, and just changed the line

```java
import com.openhtmltopdf.simple.Graphics2DRenderer;
```

to

```java
import org.xhtmlrenderer.simple.Graphics2DRenderer;
```

so the example code is using the ```Graphics2DRenderer``` from the latest Flying Saucer lib ('flying-saucer-core-9.1.1.jar'). 

Running the code again we get the following image:

![image](https://cloud.githubusercontent.com/assets/23582335/21961864/7306d32a-db15-11e6-9896-3d1240375c11.png)

Which is the expected result.

This seems to be a bug in Flying Saucer. Making it short, I have two questions:

1. Which Flying Saucer version is OpenHtmlToPdf based on?

2. Is it possible that you update the Flying Saucer code to fix the above bug (if it IS a bug)?

Thanks a lot!

**Comment by danfickle** [comment]
Hi @skjardenCode 
As we've worked on the PDF side quite a lot of bitrot has occurred in the java2d output to the point it is not currently useable. I did remove reference to image output from the readme but I should have explicitly said that image output is broken. I apologise for your wasted effort in trying to get it to work.

I have not deleted the image output code as at some point we may want to get it working again. As for your bug, it may be related to encoding or fonts. I'd recommend keep using flying saucer for image output if possible.
